### PR TITLE
Further cleanup the sync script

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Cleanup typos, simplify logic in a few places.
Make the list of files config a dict instead of a list of dicts.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

/kind cleanup

### CODE REVIEW

File: sync/test_sync.py [hunk 1]
```diff
@@ -206,13 +181,13 @@ def test_transform_text(self):
         tmp_name = None
 
         # write to file
-        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        with tempfile.NamedTemporaryFile(dir='/tmp', delete=False) as tmp:
```
Comment (afrittoli) [inline sync/test_sync.py:184]: Assuming `/tmp` is the temp folder is not portable, so I set the base dir explicitly to make the test run on mac.

File: sync/test_sync.py [hunk 2]
```diff
@@ -120,44 +119,20 @@ def test_get_files(self):
         expected = None
         actual = None
 
-        with tempfile.NamedTemporaryFile(delete=True) as tmp:
+        with tempfile.NamedTemporaryFile(dir='/tmp', delete=True) as tmp:
```
Comment (afrittoli) [inline sync/test_sync.py:122]: Assuming `/tmp` is the temp folder is not portable, so I set the base dir explicitly to make the test run on mac.

File: sync/test_sync.py [hunk 3]
```diff
@@ -105,7 +104,7 @@ def test_yaml_files_to_dic_list(self):
         actual = None
         tmp_name = None
 
-        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        with tempfile.NamedTemporaryFile(dir='/tmp', delete=False) as tmp:
```
Comment (afrittoli) [inline sync/test_sync.py:107]: Assuming `/tmp` is the temp folder is not portable, so I set the base dir explicitly to make the test run on mac.

File: sync/sync.py [hunk 4]
```diff
@@ -253,6 +227,9 @@ def create_resource(dest_prefix, file, versions):
         resource = resource_template.render(component_versions_json=serialize)
     elif ".md" in file:
```
Comment (popcor255) [inline sync/sync.py:228]: I forgot to do fix this as well. to use `endswith` instead of `in`

Comment (popcor255) [reply sync/sync.py:228]: Thanks @afrittoli 
/lgtm

Comment (afrittoli) [reply sync/sync.py]: Nice catch, thanks. Fixed.

**Review by AlanGreene — APPROVED** [review]
Thanks @afrittoli 
/lgtm

**Pull request merged**